avocado/utils/download.py

```diff
@@ -162,7 +162,7 @@ def _verify_hash(filename):
     download_failures = 0
     hash_file = _verify_hash(dst)
 
-    while not hash_file == hash_expected:
+    while not crypto.cmp_hash(hash_file, hash_expected):
```

Comment:
@dmonakhov The asset improvements pull request was merged in https://github.com/avocado-framework/avocado/pull/1339. Please let me know if it solves your issues.